crates/hdk3/src/host_fn/trace.rs

```diff
@@ -0,0 +1,79 @@
+use crate::prelude::*;
+use std::fmt::Write;
+use std::sync::atomic::AtomicUsize;
+use std::sync::atomic::Ordering;
+
+#[derive(Default)]
+pub struct WasmSubscriber {
+    ids: AtomicUsize,
+}
+
+struct StringVisitor<'a> {
+    fields: &'a mut String,
+    message: &'a mut String,
+}
+
+impl<'a> tracing_core::field::Visit for StringVisitor<'a> {
+    fn record_debug(&mut self, field: &tracing_core::Field, value: &dyn std::fmt::Debug) {
+        if field.name() == "message" {
+            let did_write = write!(self.message, "{:?}", value);
+            if did_write.is_err() {
+                let _ = write!(self.message, "**failed to write message**");
+            }
+        } else {
+            let did_write = write!(self.fields, "{} = {:?}; ", field.name(), value);
+            if did_write.is_err() {
+                let _ = write!(self.fields, "**failed to write {}**", field.name());
+            }
+        }
+    }
+}
+
+/// By implementing WasmSubscriber we integrate the rust tracing crate with the __trace host_fn without inventing some DIY DSL
+///
+/// Currently supports all the event macros for tracing such as `trace!`, `info!`, `debug!`, `warn!`, `error!`.
+///
+/// Does NOT support spans, so attempting to `#[instrument]` a function or similar will panic the wasm.
```

Comment:
Why do we want to panic the wasm? Can we just ignore spans for now, and start including the data when/if we choose to?